code.cpp

```diff
@@ -1,116 +1,120 @@
+            table.resize(table_size);
+        };
+        void random() {
+            std::random_device rd;  
+            std::mt19937 gen(rd());
+            std::uniform_int_distribution<> distrib(0, 1);
+            for (int i = 0; i < table_size; i++) {
+                table[i] = distrib(gen);
             }
-            if (i + 1 < Table::height && j > 0 && arr[Table::width * (i + 1) + (j - 1)] == 1) {
-                count++;
-            }
-            if (i + 1 < Table::height && arr[Table::width * (i + 1) + j] == 1) {
-                count++;
-            }
-            if (i + 1 < Table::height && j + 1 < Table::width && arr[Table::width * (i + 1) + (j + 1)] == 1) {
-                count++;
-            }
-            if (arr[Table::width * i + j] == 0 && count == 3){
-                res[Table::width * i + j] = 1;
-            } else if (arr[Table::width * i + j] == 1 && (count == 2 || count == 3)){
-                res[Table::width * i + j] = 1;
-            } else {
-                res[Table::width * i + j] = 0;
+        }
+        void step() {
+            std::vector <int> res(table_size); 
+            for (int i = 0; i < height; i++) {
+                for (int j = 0; j < width; j++) {
+                    int count = 0;
+                    count += (i > 0 && j > 0 && table[width * (i - 1) + (j - 1)] == 1);
+                    count += (i > 0 && table[width * (i - 1) + j] == 1);
+                    count += (i > 0 && j + 1 < width && table[width * (i - 1) + (j + 1)] == 1);
+                    count += (j > 0 && table[width * i + (j - 1)] == 1);
+                    count += (j + 1 < width && table[width * i + (j + 1)] == 1);
+                    count += (i + 1 < height && j > 0 && table[width * (i + 1) + (j - 1)] == 1);
+                    count += (i + 1 < height && table[width * (i + 1) + j] == 1);
+                    count += (i + 1 < height && j + 1 < width && table[width * (i + 1) + (j + 1)] == 1);
+                    if (table[width * i + j] == 0 && count == 3) {
+                        res[width * i + j] = 1;
+                    } else if (table[width * i + j] == 1 && (count == 2 || count == 3)) {
+                        res[width * i + j] = 1;
+                    } else {
+                        res[width * i + j] = 0;
+                    }
+                }
             }
+            table = res;  
         }
-    }
-    return res; 
-}
-
-bool is_alive(std::array<int,Table::table_size>& arr) {
-    std::array<int,Table::table_size> null_array = {};
-    if (arr != null_array) {
-        return true;
-    }
-    return false;
-}
-
-bool no_period(std::array<int,Table::table_size>& arr, std::list<std::array<int,Table::table_size>> mass) {
-    while (mass.size() != 0) {
-        if (mass.front() == arr) {
-            return false;
+        bool is_alive() {
+            std::vector<int> null_vector = {};
```

Comment:
I think you need std::vector<int>(table_size) (this with zeros) not {} (this is empty),

Reply:
Also this function is const. If a function doesn't change any variable, it can be marked as const. The reason is that const objects can only call const methods

Reply:
And come other functions are const